examples/robot_brain_cmaes/main.py

```diff
@@ -75,5 +78,16 @@ def main() -> None:
         generation_index += 1
 
 
+def get_config() -> ModuleType:
```

Comment:
So while I appreciate the effort and understand why you do this, I am hesitant to do this as it might confuse students using the examples. I think we should write the examples in a way that we want the student to write and understand them. Integration test is very nice, but  I think we should first of all look at what the student reads, if you understand me

Reply:
Understood, I just had to make this function in order to override the default config (via pytest-mock) for the examples (while keeping mypy / the validation CI happy)

Other alternatives I can think of for overriding the config:
* change `main()` to take an optional parameter like `def main(config: Optional[Dict] = None)`, and if it's not provided then the function could fallback to config.py...
  * also config.py could export a Dict object instead of being of the ModuleType...

* the unit test could temporarily change `config.py` on disk (moving it to a temp file), and then swapping it back after the test. But this feels way too hacky and if the user kills the test in the middle it will have changed the state of their local codebase.

Let me know if you have a preference for a particular approach or another idea!

Reply:
What is your opinion on the following?

Ah, understood! When you're talking about `config.py`, you're still referring to a module in Python terms, but it's a module defined by a local file rather than something you'd install through pip or similar.

If `config.py` contains constants and you want to mock them for a test, you can still use the `unittest.mock` module in combination with `pytest`. 

Let's assume you have the following structure:

```
project/
|-- main.py
|-- config.py
|-- tests/
    |-- test_main.py
```

If `main.py` imports `config.py` like so:

```python
# main.py
from config import SOME_CONSTANT
```

You can mock `SOME_CONSTANT` in your test in `test_main.py` like this:

```python
from unittest.mock import patch

def test_some_function():
    with patch('main.SOME_CONSTANT', new='Mocked Value'):
        # Here, SOME_CONSTANT will have the value 'Mocked Value' when used in main.py
        # Your test code and assertions go here
```

Or if you're using the `pytest-mock` plugin:

```python
def test_some_function(mocker):
    mocker.patch('main.SOME_CONSTANT', new='Mocked Value')
    # Your test code and assertions go here
```

Remember to patch at the point where the module or attribute is used, not where it's defined. This is a common gotcha with mocking in Python. In the above example, we patched `main.SOME_CONSTANT` because it's the `SOME_CONSTANT` from `config.py` but as used in `main.py`.

Reply:
Slightly adjusted but basically the same

Got it! If `main` imports `config` in the manner you've described and accesses constants like `config.CONSTANT`, then you'll need to slightly adjust the patching approach.

Here's how you can mock `CONSTANT` in `test_main.py`:

```python
from unittest.mock import patch

def test_some_function():
    with patch('main.config.CONSTANT', new='Mocked Value'):
        # Here, config.CONSTANT will have the value 'Mocked Value' when used in main.py
        # Your test code and assertions go here
```

If you're using the `pytest-mock` plugin:

```python
def test_some_function(mocker):
    mocker.patch('main.config.CONSTANT', new='Mocked Value')
    # Your test code and assertions go here
```

Since `main` imports `config` and uses it like `config.CONSTANT`, you mock the attribute as `main.config.CONSTANT` in your tests. Remember, always patch the attribute at the point it's used (in this case, within `main`).